include/polo/utility/custom_logger.hpp

```diff
@@ -0,0 +1,30 @@
+#ifndef POLO_UTILITY_CUSTOMLOG_HPP_
+#define POLO_UTILITY_CUSTOMLOG_HPP_
+
+namespace polo {
+namespace utility {
+namespace logger {
+template <class value_t, class index_t> struct custom {
+  using func_t = bool *(*)(const index_t, const value_t, const value_t *,
+                           const value_t *, const value_t *, void *);
+
+  custom(func_t log, void *data) {
+    log_ = log;
+    data_ = data;
+  }
+
+  template <class InputIt1, class InputIt2>
```

Comment:
I will remove the `InputIt1` and `InputIt2` here, as you would like to pass the pointers directly to `log_`.